### PR TITLE
Fix routes that need authentication

### PR DESCRIPTION
**Pull request by RvanB**

### CODE REVIEW

File: client/components/imports/WithRouter.jsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        return (
+        <Component
+            {...props}
+            router={{ location, navigate, params }}
+        />
+        );
+    }
+
+    return ComponentWithRouterProp;
+}
```
Comment (cabeese) [inline client/components/imports/WithRouter.jsx]: nit:newline

File: client/components/public/password-reset.jsx [hunk 2]
```diff
@@ -63,10 +64,13 @@ PasswordReset = class PasswordReset extends Component {
     }
 
     // Call Meteor method to create account.
-    Accounts.resetPassword(this.props.params.token, newPassword, (error, result) => {
+    Accounts.resetPassword(this.props.router.params.token, newPassword, (error, result) => {
+      console.log(this.props);
```
Comment (cabeese) [inline client/components/public/password-reset.jsx]: nit: remove

Comment (RvanB) [reply client/components/public/password-reset.jsx]: Fixed

File: client/components/imports/WithRouter.jsx (hunk 1 above)
Comment (RvanB) [reply client/components/imports/WithRouter.jsx]: Fixed

**Review by cabeese — APPROVED** [review]

**Pull request merged**